src/IndexingWrapper.cpp

```diff
@@ -242,40 +242,32 @@ indexWrapper(Rcpp::CharacterVector & dbpath, Rcpp::DataFrame & dataFrame, Rcpp::
 	    std::string val;
 
 	    if (list.containsElementNamed("slot")) {
-		try {
 		    slot = Rcpp::as<int>(list["slot"]);
```

Comment:
Removed all those commits except the first one. Used a common format for
all the name/index checks in indexWrapper. Could you check if it's okay now?

On Mon, Aug 15, 2016 at 2:12 PM, James Aylett <notifications@github.com>
wrote:

> In src/IndexingWrapper.cpp
> <https://github.com/amandaJayanetti/RXapian/pull/28#discussion_r74733837>:
>
> > @@ -242,40 +242,32 @@ indexWrapper(Rcpp::CharacterVector & dbpath, Rcpp::DataFrame & dataFrame, Rcpp::
> >  	    std::string val;
> >
> >  	    if (list.containsElementNamed("slot")) {
> > -		try {
> >  		    slot = Rcpp::as<int>(list["slot"]);
>
> What happens if someone puts something in slot that isn't an int? Having
> the try…catch before was presumably needed to deal with that case, so I'm
> not sure this is an improvement (and similarly throughout this commit).
>
> The commit message is poor as well. Is it improving readability? It seems
> to be removing functionality. If it's genuinely unneeded functionality,
> then explain why in the commit message so others reading through can see
> what's going on. (Or you, at some point in the future, trying to remember
> why you made this change!)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/amandaJayanetti/RXapian/pull/28/files/c3920af6ce7cb58b226c3a8bb1480382cbac009d..f13c1cbfce83084a2c2c4bfc2bdfbb29ba02365d#r74733837>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH_jNvDRPxy0pqsnSorR16jQqENuveR3ks5qgCZggaJpZM4Jj8cQ>
> .
>